packages/api/ai/srcbook-generator.mts

```diff
@@ -19,9 +24,17 @@ const makeGenerateCellSystemPrompt = (language: CodeLanguageType) => {
 };
 
 const makeGenerateCellUserPrompt = (session: SessionType, insertIdx: number, query: string) => {
-  const inlineSrcbookWithPlaceholder = encode(session.cells, session.metadata, {
+  // Make sure we copy cells so we don't mutate the session
+  const cellsWithPlaceholder: CellWithPlaceholderType[] = [...session.cells];
+
+  cellsWithPlaceholder.splice(insertIdx, 0, {
```

Comment:
Since `CellWithPlaceholderType` is a superset of the `CellType`, any existing calls or cell representations will succeed just fine, but here, in the one place we need to do something different for AI, we can leverage this superset type and pass in the cell instead of relying on the encoder to know what to do for AI.